File: test/aidbox_sdk/snapshots/python/hl7_fhir_r4_core/ClinicalImpression.py
from typing import Optional, List
from dataclasses import dataclass, field
from base import Annotation
from base import Period
from base import CodeableConcept
from base import DomainResource
from base import Reference
from base import Identifier
from base import BackboneElement

@dataclass(kw_only=True)
class ClinicalImpressionInvestigation(BackboneElement):
    code: CodeableConcept
    item: Optional[List[Reference]] = field(default_factory=list)

@dataclass(kw_only=True)
class ClinicalImpressionFinding(BackboneElement):
    basis: Optional[str] = None
    item_reference: Optional[Reference] = None
    item_codeable_concept: Optional[CodeableConcept] = None

@dataclass(kw_only=True)
class ClinicalImpression(DomainResource):
    description: Optional[str] = None
    date: Optional[str] = None
    investigation: Optional[List[ClinicalImpressionInvestigation]] = field(default_factory=list)
    protocol: Optional[List[str]] = field(default_factory=list)
    assessor: Optional[Reference] = None
    supporting_info: Optional[List[Reference]] = field(default_factory=list)
    encounter: Optional[Reference] = None
    problem: Optional[List[Reference]] = field(default_factory=list)
    status_reason: Optional[CodeableConcept] = None
    note: Optional[List[Annotation]] = field(default_factory=list)
    summary: Optional[str] = None
    effective_date_time: Optional[str] = None
    prognosis_codeable_concept: Optional[List[CodeableConcept]] = field(default_factory=list)
    status: str
    previous: Optional[Reference] = None
    code: Optional[CodeableConcept] = None
    identifier: Optional[List[Identifier]] = field(default_factory=list)
    finding: Optional[List[ClinicalImpressionFinding]] = field(default_factory=list)
    prognosis_reference: Optional[List[Reference]] = field(default_factory=list)
    subject: Reference
    effective_period: Optional[Period] = None

File: test/aidbox_sdk/snapshots/dotnet/hl7-fhir-r4-core/ClinicalImpression.cs
using Aidbox.FHIR.Base;
using Aidbox.FHIR.Utils;

namespace Aidbox.FHIR.R4.Core;

public class ClinicalImpression : DomainResource, IResource
{
    public string? Description { get; set; }
    public string? Date { get; set; }
    public ClinicalImpression_Investigation[]? Investigation { get; set; }
    public string[]? Protocol { get; set; }
    public Base.ResourceReference? Assessor { get; set; }
    public Base.ResourceReference[]? SupportingInfo { get; set; }
    public Base.ResourceReference? Encounter { get; set; }
    public Base.ResourceReference[]? Problem { get; set; }
    public Base.CodeableConcept? StatusReason { get; set; }
    public Base.Annotation[]? Note { get; set; }
    public string? Summary { get; set; }
    public string? EffectiveDateTime { get; set; }
    public Base.CodeableConcept[]? PrognosisCodeableConcept { get; set; }
    public required string Status { get; set; }
    public object? Effective    
    {
        get
        {
            if (EffectiveDateTime is not null)
            {
                return EffectiveDateTime;
            }
    
            if (EffectivePeriod is not null)
            {
                return EffectivePeriod;
            }
    
            return null;
        }
    
        set
        {
            if (value?.GetType() == typeof(string))
            {
                EffectiveDateTime = (string)value;return;
            }
    
            if (value?.GetType() == typeof(Base.Period))
            {
                EffectivePeriod = (Base.Period)value;return;
            }
    
            throw new ArgumentException("Invalid type provided");
        }
    }
    public Base.ResourceReference? Previous { get; set; }
    public Base.CodeableConcept? Code { get; set; }
    public Base.Identifier[]? Identifier { get; set; }
    public ClinicalImpression_Finding[]? Finding { get; set; }
    public Base.ResourceReference[]? PrognosisReference { get; set; }
    public required Base.ResourceReference Subject { get; set; }
    public Base.Period? EffectivePeriod { get; set; }

    public class ClinicalImpression_Investigation : BackboneElement
    {
        public required Base.CodeableConcept Code { get; set; }
        public Base.ResourceReference[]? Item { get; set; }
    }

    public class ClinicalImpression_Finding : BackboneElement
    {
        public string? Basis { get; set; }
        public Base.ResourceReference? ItemReference { get; set; }
        public Base.CodeableConcept? ItemCodeableConcept { get; set; }
    }
}